apps/aechannel/src/aesc_fsm.erl

```diff
@@ -259,11 +260,14 @@ upd_deposit(Fsm, #{amount := Amt} = Opts) when is_integer(Amt) ->
     lager:debug("upd_deposit(~p)", [Opts]),
     gen_statem:call(Fsm, {upd_deposit, Opts}).
 
-upd_transfer(_Fsm, _From, _To, Amount) when Amount < 0 ->
+upd_transfer(Fsm, From, To, Amount) ->
+    upd_transfer(Fsm, From, To, Amount, #{}).
+
+upd_transfer(_Fsm, _From, _To, Amount, _Opts) when Amount < 0 ->
```

Comment:
Possibly unrelated: Why would a transfer of 0 make sense?

Reply:
This is a funny edge case. I don't see a reason of forbidding it, nor do I see any meaningful use of it.

Reply:
Well, it doesn't break anything, so why would we disallow it? ;-)